packages/grafana-ui/src/components/VizLayout/VizLayout.story.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { withCenteredStory } from '../../utils/storybook/withCenteredStory';
+import { number } from '@storybook/addon-knobs';
+import { VizLayout } from './VizLayout';
+
+export default {
+  title: 'Visualizations/VizLayout',
+  component: VizLayout,
+  decorators: [withCenteredStory],
+  parameters: {
+    docs: {},
+  },
+};
+
+const getKnobs = () => {
+  return {
+    legendWidth: number('legendWidth', 100),
+    legendItems: number('legendItems', 2),
+  };
+};
+
+export const BottomLegend = () => {
+  const { legendItems } = getKnobs();
+  const items = Array.from({ length: legendItems }, (_, i) => i + 1);
+
+  const legend = (
+    <VizLayout.Slot position="bottom" maxHeight="30%">
+      {items.map((_, index) => (
+        <div style={{ height: '30px', width: '100%', background: 'blue', marginBottom: '2px' }}>
+          Legend item {index}
+        </div>
+      ))}
+    </VizLayout.Slot>
+  );
+
+  return (
+    <VizLayout width={400} height={500} legend={legend}>
```

Comment:
I think the "slots" (legend in this case) might be independent from the props and rendered declaratively instead? Don't think VizLayout component API should know any specifics about what's there to be rendered

Reply:
Not sure what you mean independent? Slots/Legend needs to be part of the API somehow either via props or via children

Having it via children makes the API A LOT more complicated, needed multiple different child components and possibly react context to access the viz size. I started with that but using props for the slots and the main child render prop for the viz made it 10x simpler and more elegant.  And it's still declarative like this, just passed via prop instead of via child 

I can change the name of the prop from legend to spots?

Reply:
>  Don't think VizLayout component API should know any specifics about what's there to be rendered

It does need to know the constraints (max-width / height) & position of the slots and that is all that is done here via the VizLayout.Slot component

Reply:
renamed VizLayout.Slot to VizLayout.Legend as it feels more accurate